frontend/public/components/utils/mask-data.jsx

```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { CopyToClipboard } from './index';
+
+export const MaskedData = () => <React.Fragment>
+  <span className="sr-only">Value hidden</span>
+  <span aria-hidden="true">&bull;&bull;&bull;&bull;&bull;</span>
+</React.Fragment>;
+
+export class MaskData extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { showSecret: false };
+    this.toggleSecret = this.toggleSecret.bind(this);
+  }
+
+  toggleSecret() {
+    this.setState({ showSecret: !this.state.showSecret });
+  }
+
+  render() {
+    const { title, value } = this.props;
+    const { showSecret } = this.state;
+    const visibleValue = showSecret ? value : <MaskedData /> ;
+
+    return <React.Fragment>
+      <dt>{title} {value &&
```

Comment:
Let's not have the `dt`/`dd` be part of this. Better to make this just the value so that it can be used outside of a `dl` element.